src/recipe.cpp

```diff
@@ -0,0 +1,361 @@
+#include "recipe.h"
+
+#include "calendar.h"
+#include "generic_factory.h"
+#include "itype.h"
+#include "output.h"
+#include "skill.h"
+
+#include <algorithm>
+
+recipe::recipe() : skill_used( "none" ) {}
+
+int recipe::batch_time( int batch, float multiplier, size_t assistants ) const
```

Comment:
Not very related, just noting:
`assistants` should eventually become more unique, so that a lazy, armless drunk is less effective than motivated expert on meth.